sdks/python/src/lilypad/_internal/versioning.py

```
@@ -65,18 +66,26 @@ class AsyncVersionedResult(AsyncTracedResult[R]):
 class _BaseVersionedFunction(_BaseTracedFunction[P, R, Any]):
     """Base class for versioned functions."""
 
-    closure: Closure = field(init=False)
+    closure: Closure | None = field(init=False, default=None)
 
     def __post_init__(self) -> None:
         super().__post_init__()
-        self.closure = Closure.from_fn(self.fn)
+        try:
+            self.closure = Closure.from_fn(self.fn)
+        except subprocess.CalledProcessError as error:
```

The `Closure.from_fn` method should raise an error rather than let the ruff formatting error propagate.

For example:

```python
Raises:
    ClosureFormattingError: If `ruff` fails to run and the code cannot be
        formatted
```

Then it will be more clear when using `Closure.from_fn` that you need to handle said failure.